x/profile/keeper/msg_server_set_validator_cons_address.go

```diff
@@ -0,0 +1,72 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	valtypes "github.com/tendermint/spn/pkg/types"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) SetValidatorConsAddress(
+	goCtx context.Context,
+	msg *types.MsgSetValidatorConsAddress,
+) (*types.MsgSetValidatorConsAddressResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	valPubKey, err := valtypes.NewValidatorConsPubKey(msg.ValidatorConsPubKey, msg.ValidatorKeyType)
+	if err != nil {
+		return &types.MsgSetValidatorConsAddressResponse{},
+			sdkerrors.Wrap(types.ErrInvalidValidatorKey, string(msg.ValidatorConsPubKey))
+	}
+	consAddress := valPubKey.GetConsAddress().String()
+
+	// cannot set the consensus key if key is used for another validator
+	validatorByConsAddr, found := k.GetValidatorByConsAddress(ctx, consAddress)
```

Comment:
I think we should allow the validator to set a new validator address for the consensus key since the signature prove they are the owner of it